### DISCUSSION TITLE
Code example on how to use soundex to block

### DISCUSSION THREAD

**Discussion by illeamb** [moj-analytical-services/splink#1110] [Q&A] [answered]
I did noticed in the  documentation regarding blocking with soundex. Is it possible to get a working sample code. 

I have tried the following : 

blocking_rule_1 = "SOUNDEX(l.surname)= SOUNDEX(r.surname)"
count = linker.count_num_comparisons_from_blocking_rule(blocking_rule_1)
print(f"Number of comparisons generated by '{blocking_rule_1}': {count:,.0f}")

I am getting the following error as it can't a function for soundex.  If you could let me know it would be great. 

File /opt/conda/lib/python3.10/site-packages/splink/duckdb/duckdb_linker.py:192, in DuckDBLinker._execute_sql_against_backend(self, sql, templated_name, physical_name)
    185 logger.log(5, log_sql(sql))
    187 sql = f"""
    188 CREATE TABLE {physical_name}
    189 AS
    190 ({sql})
    191 """
--> 192 self._con.execute(sql).fetch_df()
    194 return DuckDBLinkerDataFrame(templated_name, physical_name, self)

CatalogException: Catalog Error: Scalar Function with name soundex does not exist!
Did you mean "round"?
LINE 18:     SOUNDEX(l.surname)= SOUNDEX(r.surname) and l.dob=r.dob
    where l."source_dataset" || '-__-' || l."unique_id" < r."source_dataset" || '-__-' || r."unique_id" and l."source_dataset" != r."source_dataset"
    )
        ...
             ^

**Comment by ADBond** [comment, marked as answer]
Unfortunately the `SOUNDEX` function is not available for the duckdb backend - you can see the available text functions in [this page of the documentation](https://duckdb.org/docs/sql/functions/char).

You will either have to switch to using Spark instead, or pre-process your data to have a `surname_soundex` column which you can then block on directly with equality (which may also have computational advantages - then you will only be calculating the soundex for each row of your data once, rather than every time it is compared with another row).
